icu4c/source/i18n/measunit_impl.h

```diff
@@ -41,8 +41,8 @@ struct SingleUnitImpl : public UMemory {
      * considered the same as square-meter.
      *
      * The dimensionless unit generally does not get compared, but if it did, it
-     * would sort before other units by virtue of dimensionality being 0 and
-     * index being < 0.
+     * would sort before other units by virtue of index being < 0 and
+     * dimensionality not being negative.
```

Comment:
I verified that dimensionless units never got compared by temporarily adding a set of asserts.

I wasn't sure whether to leave those asserts in there... I chose not to, because it's not fundamentally necessary: code could go compare with them if it wants.

Reply:
OK.  Fine to leave as-is